spec/requests/posts_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  context 'status is succesful ' do
+    it 'index returns a success response' do
+      get '/users/1/posts'
+      expect(response).to be_successful
+    end
+  end
+  context 'the right place holder text is rendered ' do
+    it 'renders all posts for a particular user on the page ' do
+      get '/users/1/posts'
+      expect(response.body).to include('The following is a list of posts')
+    end
+
+    it 'renders all the specifics of a single post based on the id ' do
+      get '/users/1/posts/3'
+      expect(response.body).to include('This is a post whose id is 3')
+    end
+  end
+
+  describe 'GET /show' do
```

Comment:
Spec file should include tests that ensure the response body includes the correct placeholder text.
Example : expect(response.body).to include("Welcome to user index page"). Kindly test for the same.